pkg/server/node.go

```diff
@@ -451,25 +451,32 @@ func (n *Node) start(
 		}
 		return nil
 	}); err != nil {
-		return errors.Wrapf(err, "failed to read last up timestamp from stores")
+		return errors.Wrapf(err, "failed to read last up timestamp from stores"), nil
 	}
 	n.lastUp = mostRecentTimestamp.WallTime
 
 	// Set the stores map as the gossip persistent storage, so that
 	// gossip can bootstrap using the most recently persisted set of
 	// node addresses.
 	if err := n.storeCfg.Gossip.SetStorage(n.stores); err != nil {
-		return fmt.Errorf("failed to initialize the gossip interface: %s", err)
+		return fmt.Errorf("failed to initialize the gossip interface: %s", err), nil
 	}
 
-	// Bootstrap any uninitialized stores.
-	//
-	// TODO(tbg): address https://github.com/cockroachdb/cockroach/issues/39415.
-	// Should be easy enough. Writing the test is probably most of the work.
+	// Bootstrap any uninitialized stores and signal channel once complete.
+	bootstrapStoresComplete := make(chan struct{})
```

Comment:
s/bootstrapStoresComplete/bootstrapNewStoresDone

Reply:
[nit] I think what we could do instead here is something like the following:
```go
var waitForBootstrapNewStores := func() { }
if len(state.newEngines) > 0 {
  bootstrapStoresComplete := make(chan struct{})
  waitForBootstrapNewStores = func() { <-bootstrapStoresComplete }
  // ...
}

// ...
return nil, waitForBootstrapNewStores
```

What do ya think? I think it localizes usage of the signal channel to just the conditional where it's needed.

Reply:
I think that's definitely better! I'll make the change.